src/physics/em/BetheHeitlerInteractor.i.hh

```diff
@@ -0,0 +1,233 @@
+            real_type epsilon_sq = epsilon * epsilon;
+            // Calculate f1 density function
+            real_type f1 = 3 / std::pow(0.5 - epsilon_min, 3)
+                           * (epsilon_sq - epsilon + 0.25);
+            // Calculate delta from element atomic number and sampled epsilon
+            real_type d =
+                this->impact_parameter(element_.atomic_number(), epsilon);
+            // Calculate g1 "rejection" function
+            reject_threshold = this->screening_phi1_aux(d, element_.atomic_number())
+                               / this->screening_phi1_aux(delta_min, element_.atomic_number());
+        }
+        else
+        {
+            // Used to sample from f2
+            epsilon = epsilon_min + (0.5 - epsilon_min) * rnd_b(rng);
+            // Calculate f2 density function (constant)
+            real_type f2 = 1 / (0.5 - epsilon_min);
+            // Calculate delta given the element atomic number and sampled epsilon
+            real_type d = this->impact_parameter(element_.atomic_number(), epsilon);
+            // Calculate g2 "rejection" function
+            reject_threshold = this->screening_phi2_aux(d, element_.atomic_number())
+                               / this->screening_phi2_aux(delta_min, element_.atomic_number());
+        }
+        CHECK(epsilon >= epsilon_min && epsilon <= 0.5);
+    } while (BernoulliDistribution(1 - reject_threshold)(rng));
+
+    // Construct interaction for change to primary (incident) particle
+    Interaction result;
+    result.action      = Action::absorbed;
+    result.energy      = units::MevEnergy{0};
+    result.direction   = {0, 0, 0};
+    result.secondaries = {electron_pair, 2};
+
+    // Argument to density function for polar angle
+    real_type log_r2r3
+        = std::log(generate_canonical(rng) * generate_canonical(rng));
+    real_type density_arg = log_r2r3
+                            * BernoulliDistribution(0.25)(rng) ? 1.6 : (1 / 1.875);
+        
+    // Polar angle from the density function
+    real_type theta = 0.0977
+                      * (density_arg * std::exp(-log_r2r3)
+                         + 27 * density_arg * std::exp(-log_r2r3));
+
+    // Outgoing secondaries are electron and positron
+    electron_pair[0].def_id = shared_.electron_id;
+    electron_pair[1].def_id = shared_.positron_id;
+    // Energies of secondaries shared equally
+    electron_pair[0].energy
+        = units::MevEnergy{0.5 * epsilon * inc_energy_.value()};
+    electron_pair[1].energy = electron_pair[0].energy;
+
+    // Azimuthal angle sampled isotropically
+    UniformRealDistribution<real_type> sample_phi(0, 2 * constants::pi);
+    electron_pair[0].direction = rotate(
+        from_spherical(std::cos(theta), sample_phi(rng)), inc_direction_);
+    // Opposite direction in costheta
+    electron_pair[1].direction = rotate(
+        from_spherical(-std::cos(theta), sample_phi(rng)), inc_direction_);
```

Comment:
Sampling an independent azimuthal angle for the two electrons doesn't look right to me here -- are you sure? I'm also a little confused how the sampled directions are modified in the loop below -- you sample exiting directions for the electrons based on the incident gamma's, and then flip that direction around by subtracting from the incident direction?

Reply:
Did you see the previous question/comment here?